### PR TITLE
Add MQIIH header support

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Related to https://github.com/ballerina-platform/ballerina-library/issues/5730
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/topic_publisher_subscriber_tests.bal [hunk 1]
```diff
@@ -428,6 +422,64 @@ function publishSubscribeWithMQCIHHeadersTest() returns error? {
...
+                securityScope: "s"
+            }
+        ]
+    });
+    Message? message = check subscriber->get();
+    if message !is () {
+        test:assertEquals(string:fromBytes(message.payload), "Hello World");
+        Header[]? headers = message.headers;
+        if headers is () {
+            test:assertFail("Expected MQCIH headers");
```
Comment (NipunaRanasinghe) [inline ballerina/tests/topic_publisher_subscriber_tests.bal:453]:
```suggestion
            test:assertFail("Expected MQIIH headers");
```

File: ballerina/types.bal [hunk 2]
```diff
@@ -263,3 +257,39 @@ public type MQCIH record {|
...
+    int encoding = 0;
+    string strucId = "IIH ";
+    int strucLength = 84;
+    int version = 1;
+    int codedCharSetId = 0;
+    string format = "        ";
+    string lTermOverride = "        ";
+    string mfsMapName = "        ";
+    string replyToFormat = "        ";
+    string authenticator = "        ";
```
Comment (NipunaRanasinghe) [inline ballerina/types.bal:290]: Can we define a common constant for the default value and reuse in here? 

```suggestion
    string format = "        ";
    string lTermOverride = "        ";
    string mfsMapName = "        ";
    string replyToFormat = "        ";
    string authenticator = "        ";
```

Comment (dilanSachi) [reply ballerina/types.bal]: Fixed

File: ballerina/tests/topic_publisher_subscriber_tests.bal (hunk 1 above)
Comment (dilanSachi) [reply ballerina/tests/topic_publisher_subscriber_tests.bal]: Fixed

**Review by ayeshLK — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**